debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java

```diff
@@ -428,7 +428,9 @@ protected CommonConnectorConfig(Configuration config, String logicalName, int de
     /**
      * Provides access to the "raw" config instance. In most cases, access via typed getters for individual properties
      * on the connector config class should be preferred.
+     * TODO this should be protected in the future to force proper facade methods based access / encapsulation
      */
+    @Deprecated
```

Comment:
I added a TODO here and a deprecation notice. We recently discussed that there should be no direct access to Confdiguration but all encapsulated with methods (PostgresConnectorConfig, MongoDBConnectorConfig, MysqlConnectorConfig, ...). I think it was just convenient in the past to just pass the Configuration instance instead of encapsulating it?

Reply:
Yes, exactly, eventually there should be no need for this method. Right now the code is in a somewhat transitioning state and no new code should make access of the raw config. When working on something it's also a good idea to update related usages; so far we haven't done a general refactoring, though.

Reply:
great, so we can keep the annotation?

Reply:
Yes.